src/probability/martingale.lean

```diff
@@ -353,6 +386,77 @@ end
 
 end submartingale
 
+-- We now prove the converse of the optional stopping theorem
+
+section piecewise_const
+
+variables {i j n : ℕ} {s : set α} {x : α} [decidable_pred (λ k, k ∈ s)]
+
+lemma piecewise_const_eq_iff :
+  s.piecewise (λ _, i) (λ _, j) x = n ↔ (x ∈ s ∧ i = n) ∨ (x ∉ s ∧ j = n) :=
+by { rw [set.piecewise], dsimp only, split_ifs; simp [h], }
```

Comment:
We already have this proof elsewhere:
```suggestion
ite_eq_iff
```
I'd lean towards just removing this lemma and using `rw [set.piecewise, ite_eq_iff]` in its place